89.md

```diff
@@ -0,0 +1,112 @@
+NIP-89
+======
+
+Recommended Application Handlers
+--------------------------------
+
+`draft` `optional` `author:pablof7z`
+
+This NIP describes `kind:31989` and `kind:31990`: a way to discover applications that can handle unknown event-kinds.
+
+## Rationale
+Nostr's discoverability and transparent event interaction is one of its most interesting/novel mechanics.
+This NIP provides a simple way for clients to discover applications that handle events of a specific kind to ensure smooth cross-client and cross-kind interactions.
+
+### Parties involved
+There are three actors to this workflow:
+
+* application that handles a specific event kind (note that an application doesn't necessarily need to be a distinct entity and it could just be the same pubkey as user A)
+    * Publishes `kind:31990`, detailing how apps should redirect to it
+* user A, who recommends an app that handles a specific event kind
+    * Publishes `kind:31989`
+* user B, who seeks a recommendation for an app that handles a specific event kind
+    * Queries for `kind:31989` and, based on results, queries for `kind:31990`
+
+# Events
+
+## Recommendation event
+```json
+{
+    "kind": 31989,
+    "pubkey": <recommender-user-pubkey>,
+    "tags": [
+        [ "d", <supported-event-kind> ],
+        [ "a", "31990:app1-pubkey:<d-identifier>", "wss://relay1", "ios" ],
+        [ "a", "31990:app2-pubkey:<d-identifier>", "wss://relay2", "web" ]
+    ]
+}
+```
+
+Multiple `a` tags can appear on on the same `kind:31989`.
+
+The second value of the tag SHOULD be a relay hint.
+The third value of the tag SHOULD be the platform where this recommendation might apply.
+
```

Comment:
Could we interpret the third value as "user A is using this app on this platform", instead of "user A recommends this app for this platform"? I think recommendations/endorsements should be a separate thing like nip-32.

Reply:
Also, if I'm using this app on several platforms, can I use forth and fifth value for "a" tag? Or maybe comma-separated list of platforms as third value?

Reply:
Better just keep it to have multiple `a` tags, even if the event it's pointing to is the same.